aspnetcore/security/cors.md

```diff
@@ -42,110 +42,97 @@ These URLs have different origins than the previous two URLs:
+See [Test CORS](#testc) for instructions on testing code similar to the preceding code.
 
-<a name="acpall"></a>
-
-### Apply CORS policies to all endpoints
-
-The following code applies CORS policies to all the apps endpoints via CORS Middleware:
-```csharp
-public void Configure(IApplicationBuilder app, IHostingEnvironment env)
-{
-    // Preceding code omitted.
-    app.UseRouting();
+The <xref:Microsoft.Extensions.DependencyInjection.MvcCorsMvcCoreBuilderExtensions.AddCors*> method call adds CORS services to the app's service container:
 
-    app.UseCors();
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Startup.cs?name=snippet2)]
 
-    app.UseEndpoints(endpoints =>
-    {
-        endpoints.MapControllers();
-    });
+For more information, see [CORS policy options](#cpo) in this document.
 
-    // Following code omitted.
-}
-```
+The <xref:Microsoft.AspNetCore.Cors.Infrastructure.CorsPolicyBuilder> method can chain methods, as shown in the following code:
 
-> [!WARNING]
-> With endpoint routing, the CORS middleware must be configured to execute between the calls to `UseRouting` and `UseEndpoints`. Incorrect configuration will cause the middleware to stop functioning correctly.
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Startup2.cs?name=snippet)]
 
-See [Enable CORS in Razor Pages, controllers, and action methods](#ecors) to apply CORS policy at the page/controller/action level.
-
-See [Test CORS](#test) for instructions on testing the preceding code.
+Note: The specified URL must **not** contain a trailing slash (`/`). If the URL terminates with `/`, the comparison returns `false` and no header is returned.
 
 <a name="ecors"></a>
 
 ## Enable Cors with endpoint routing
 
-With endpoint routing, CORS can be enabled on a per-endpoint basis using the `RequireCors` set of extension methods.
+Enabling CORS on a per-endpoint basis using `RequireCors` currently does ***not*** support [automatic pre-flight requests](#apf). For more information, see [this GitHub issue](https://github.com/dotnet/aspnetcore/issues/18665) and [Test CORS with endpoint routing and [HttpOptions]](#tcer).
 
-```csharp
-app.UseEndpoints(endpoints =>
-{
-  endpoints.MapGet("/echo", async context => context.Response.WriteAsync("echo"))
-    .RequireCors("policy-name");
-});
+With endpoint routing, CORS can be enabled on a per-endpoint basis using the <xref:Microsoft.AspNetCore.Builder.CorsEndpointConventionBuilderExtensions.RequireCors*> set of extension methods:
 
-```
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/StartupEndPt.cs?name=snippet)]
 
-Similarly, CORS can also be enabled for all controllers:
+In the preceding code:
 
-```csharp
-app.UseEndpoints(endpoints =>
-{
-  endpoints.MapControllers().RequireCors("policy-name");
-});
-```
+* `app.UseCors` enables the CORS middleware but doesn't add a policy. `app.UseCors` by itself, without a policy, does ***not*** allow cross-origin requests.
```

Comment:
This might need a little more clarification. For example, if we use `AddDefaultPolicy` in the `AddCors` setup, `UseCors()` enables CORS for that policy, as the default.